src/unity/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -127,11 +127,22 @@ gl_sframe augment_data(const gl_sframe &data,
                        const std::string& image_column_name,
                        const std::string& target_column_name,
                        const gl_sarray &backgrounds,
-                       long long seed) {
+                       long long seed,
+                       bool verbose) {
+  size_t backgrounds_size = backgrounds.size();
+  size_t total_augmented_rows = data.size() * backgrounds_size;
+  table_printer table( { {"Images Augmented", 0}, {"Elapsed Time", 0}, {"Percent Complete", 0} } );
+  if (verbose) {
+    logprogress_stream << "Augmenting input images using " << backgrounds.size() << " background images." << std::endl;
+    table.print_header();
+  }
+
   auto column_index_map = generate_column_index_map(data.column_names());
   std::vector<flexible_type> annotations;
   std::vector<flexible_type> images;
+  int input_row_index = -1;
   for (const auto& row: data.range_iterator()) {
```

Comment:
We’re iterating over all the rows in a single for loop? This really should use `gl_sframe::apply`. Would be 8-16 times faster on most machines.

Probably too late for 5.6 though. Filed #2023